components/processor/gcomapiprocessor/internal/gcom/cache/auth_cache.go

```diff
@@ -0,0 +1,96 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/opentelemetry-collector-components/processor/gcomapiprocessor/internal/gcom/client"
```

Comment:
we might need a linter to enforce three import groups -- this one here would belong to a third group, of local imports